synapse/storage/databases/main/events.py

```diff
@@ -130,7 +130,7 @@ async def _persist_events_and_state_updates(
         *,
         current_state_for_room: Dict[str, StateMap[str]],
         state_delta_for_room: Dict[str, DeltaState],
-        new_forward_extremeties: Dict[str, List[str]],
+        new_forward_extremities: Dict[str, Set[str]],
```

Comment:
Could these be `Collection` instead of `Set`? They don't seem to actually need to be sets? (Same for `_persist_events_txn`.)

Mostly... is there any bug here or was it just some incorrect type hints?

Reply:
Good question! I'd prefer to keep these as `Set`s, since `new_forward_extremities` eventually makes it down to `_update_forward_extremities_txn`, which relies upon there being no duplicates.

The `List` was just an incorrect type hint which got flagged once we typed the return of `_calculate_new_extremities`.

Reply:
Makes sense, just wasn't sure if we were overly strictly typing. 👍